client/api/preview.js

```diff
@@ -0,0 +1,10 @@
+import request from './request';
+
+function createPreview(courseId, activityId) {
+  return request.get(`courses/${courseId}/activities/${activityId}/preview`)
+    .then(res => window.open(res.data));
```

Comment:
Opening new window shouldn't be part of api method ⚠️ 
Move it to callsite instead :bowtie: